src/components/CreateEditNoteContainer/NoteForm/NoteForm.tsx

```diff
@@ -0,0 +1,65 @@
+import Button from '@mui/material/Button';
+import { useFormik } from 'formik';
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router';
+import { useParams } from 'react-router-dom';
+
+import { addNote, editNote } from '../../../redux/notes-slice';
+import { AppStateType } from '../../../redux/redux-store';
+import styles from './NoteForm.module.scss';
+
+const NoteForm: React.FC<{ isEdit: boolean }> = ({ isEdit }) => {
+  const dispatch = useDispatch();
+  const { noteId } = useParams<{ noteId: string }>() || undefined;
+  const note = useSelector((state: AppStateType) =>
+    state.notesPage.notes.find((note) => noteId && note.id === +noteId)
+  );
+  const navigate = useNavigate();
+  const initialFormValues =
+    isEdit && note !== undefined
+      ? {
+          id: `${noteId}`,
+          name: `${note.name}`,
+          content: `${note.content}`,
+        }
+      : {
+          name: '',
+          content: '',
+        };
+
+  const { values, handleChange, handleSubmit, setSubmitting, isSubmitting, dirty } = useFormik({
+    initialValues: initialFormValues,
+
+    onSubmit: (values) => {
+      setTimeout(() => {
```

Comment:
Why do you use setTimeout here?

Reply:
oh, I guess you just imitating async request to server this way ?

Reply:
for animation and smoothness, I thought it would be better

Reply:
remove it later when server will be integrated